subsys/bluetooth/shell/gatt.c

```diff
@@ -108,13 +111,25 @@ static int cmd_exchange_mtu(const struct shell *sh,
 		return -ENOEXEC;
 	}
 
+	if (exchange_params.func) {
+		shell_print(sh, "Shell command busy. A previous invocation is in progress.");
+		return -EAGAIN;
+	}
+	exchange_params.func = exchange_func;
+
 	err = bt_gatt_exchange_mtu(default_conn, &exchange_params);
+
+	if (err) {
+		/* Release global `exchange_params`. */
+		exchange_params = (struct bt_gatt_exchange_params){};
```

Comment:
Given that you only change the `func`, why not just
`exchange_params.func = NULL`?

Reply:
I'll add you suggestion because I'm tired of this PR and just want to get it in.

The reason I zero out the whole struct is 1. I don't want to make assumptions about whether `bt_gatt_exchange_mtu` messed up `exchange_params`. It's not passed in as const. 2. The less some code knows about the circumstances it's in, the easier it's to follow. I think it's better if this line of code can be understood without reading the rest of the function. That "we only change `func`" can be made irrelevant and not occupy thinking resources.